ocaml-lsp-server/src/base_formatter.ml

```diff
@@ -0,0 +1,49 @@
+open Import
+
+let read_to_end (in_chan : in_channel) : string =
+  let buf = Buffer.create 0 in
+  let chunk_size = 1024 in
+  let chunk = Bytes.create chunk_size in
+  let rec go pos =
+    let actual_len = input in_chan chunk pos chunk_size in
+    if actual_len > 0 then (
+      Buffer.add_subbytes buf chunk 0 actual_len;
+      go pos
+    )
+  in
+  go 0;
+  Buffer.contents buf
+
+type command_result =
+  { stdout : string
+  ; stderr : string
+  ; status : Unix.process_status
+  }
+
+let run_command command ?stdin_value args : command_result =
+  let command =
+    match args with
+    | [] -> command
+    | _ -> Printf.sprintf "%s %s" command (String.concat ~sep:" " args)
```

Comment:
Would it be better to use `Unix.open_process_args_full`, rather than appending the arguments together? I could imagine this being a problem if the filename has spaces (as `ocamlformat --name=a folder/file.ml` is incorrect).

I realise this was in the code before PR, so may have been discussed already!

Reply:
Good call. I suppose we accidentally omitted it in the last PR.